openjdk-12 fails to build on amd64

The package openjdk-12 with version 12.0.2.10 and epoch 7 fails to build from source on amd64 with the following error:

```
2025-06-02 07:36:30.602642587 +0000 UTC cc1plus: warning: command-line option '-Wno-int-conversion' is valid for C/ObjC but not for C++
2025-06-02 07:36:30.693457647 +0000 UTC cc1plus: warning: command-line option '-Wno-int-conversion' is valid for C/ObjC but not for C++
2025-06-02 07:36:30.810417333 +0000 UTC cc1plus: warning: command-line option '-Wno-int-conversion' is valid for C/ObjC but not for C++
2025-06-02 07:36:30.825431954 +0000 UTC cc1plus: warning: command-line option '-Wno-int-conversion' is valid for C/ObjC but not for C++
2025-06-02 07:36:30.913409305 +0000 UTC cc1plus: warning: command-line option '-Wno-int-conversion' is valid for C/ObjC but not for C++
2025-06-02 07:36:31.301424075 +0000 UTC Compiling 15 files for jdk.attach
2025-06-02 07:36:31.660713513 +0000 UTC /home/build/src/java.base/unix/native/libnet/DefaultProxySelector.c: In function 'getProxyByGProxyResolver':
2025-06-02 07:36:31.661068569 +0000 UTC /home/build/src/java.base/unix/native/libnet/DefaultProxySelector.c:375:16: error: too many arguments to function 'g_proxy_resolver_lookup'; expected 0, have 4
2025-06-02 07:36:31.661293604 +0000 UTC   375 |     proxies = (*g_proxy_resolver_lookup)(resolver, uri, NULL, &error);
2025-06-02 07:36:31.661402089 +0000 UTC       |               ~^~~~~~~~~~~~~~~~~~~~~~~~~ ~~~~~~~~
2025-06-02 07:36:31.66162694 +0000 UTC /home/build/src/java.base/unix/native/libnet/DefaultProxySelector.c:394:34: error: too many arguments to function 'g_network_address_parse_uri'; expected 0, have 3
2025-06-02 07:36:31.661672641 +0000 UTC   394 |                                 (*g_network_address_parse_uri)(proxies[i], 0,
2025-06-02 07:36:31.66170579 +0000 UTC       |                                 ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ~~~~~~~~~~
2025-06-02 07:36:31.661712924 +0000 UTC /home/build/src/java.base/unix/native/libnet/DefaultProxySelector.c:399:38: error: too many arguments to function 'g_network_address_get_hostname'; expected 0, have 1
2025-06-02 07:36:31.661719189 +0000 UTC   399 |                             phost = (*g_network_address_get_hostname)(conn);
2025-06-02 07:36:31.66172417 +0000 UTC       |                                     ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ~~~~
2025-06-02 07:36:31.6617296 +0000 UTC /home/build/src/java.base/unix/native/libnet/DefaultProxySelector.c:400:38: error: too many arguments to function 'g_network_address_get_port'; expected 0, have 1
2025-06-02 07:36:31.661734746 +0000 UTC   400 |                             pport = (*g_network_address_get_port)(conn);
2025-06-02 07:36:31.661740149 +0000 UTC       |                                     ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~ ~~~~
2025-06-02 07:36:31.662530299 +0000 UTC /home/build/src/java.base/unix/native/libnet/DefaultProxySelector.c:436:10: error: too many arguments to function 'g_strfreev'; expected 0, have 1
2025-06-02 07:36:31.662701536 +0000 UTC   436 |         (*g_strfreev)(proxies);
2025-06-02 07:36:31.662866336 +0000 UTC       |         ~^~~~~~~~~~~~ ~~~~~~~
2025-06-02 07:36:31.89741789 +0000 UTC /home/build/src/java.base/share/native/libjli/java.c: In function 'TranslateApplicationArgs':
2025-06-02 07:36:31.897734123 +0000 UTC /home/build/src/java.base/share/native/libjli/java.c:1700:35: warning: the comparison will always evaluate as 'false' for the pointer operand in 'arg + 2' must not be NULL [-Waddress]
2025-06-02 07:36:31.89787186 +0000 UTC  1700 |             *nargv++ = ((arg + 2) == NULL) ? NULL : JLI_StringDup(arg + 2);
2025-06-02 07:36:31.897951501 +0000 UTC       |                                   ^~
2025-06-02 07:36:31.917453494 +0000 UTC /home/build/src/java.base/unix/native/libjli/java_md_solinux.c: In function 'ContinueInNewThread0':
2025-06-02 07:36:31.917513066 +0000 UTC /home/build/src/java.base/unix/native/libjli/java_md_solinux.c:747:37: warning: cast between incompatible function types from 'int (*)(void *)' to 'void * (*)(void *)' [-Wcast-function-type]
2025-06-02 07:36:31.917523104 +0000 UTC   747 |     if (pthread_create(&tid, &attr, (void *(*)(void*))continuation, (void*)args) == 0) {
2025-06-02 07:36:31.917530334 +0000 UTC       |                                     ^
2025-06-02 07:36:31.958500487 +0000 UTC make[3]: *** [Lib-java.base.gmk:44: /home/build/build/linux-x86_64-server-release/support/native/java.base/libnet/DefaultProxySelector.o] Error 1
2025-06-02 07:36:31.962389059 +0000 UTC make[3]: *** Waiting for unfinished jobs....
2025-06-02 07:36:32.990017909 +0000 UTC /home/build/src/java.base/share/native/libjli/parse_manifest.c: In function 'find_positions':
2025-06-02 07:36:32.990386726 +0000 UTC /home/build/src/java.base/share/native/libjli/parse_manifest.c:292:34: warning: pointer 'endpos' used after 'free' [-Wuse-after-free]
2025-06-02 07:36:32.990534802 +0000 UTC   292 |             pos = flen - (endpos - cp);
2025-06-02 07:36:32.990631531 +0000 UTC       |                          ~~~~~~~~^~~~~
2025-06-02 07:36:32.990854477 +0000 UTC /home/build/src/java.base/share/native/libjli/parse_manifest.c:291:13: note: call to 'free' here
2025-06-02 07:36:32.991011991 +0000 UTC   291 |             free(buffer);
2025-06-02 07:36:32.991091583 +0000 UTC       |             ^~~~~~~~~~~~
2025-06-02 07:36:33.357438899 +0000 UTC Compiling 400 files for jdk.compiler
2025-06-02 07:36:35.199430066 +0000 UTC /home/build/src/java.base/unix/native/libnet/NetworkInterface.c: In function 'addif':
2025-06-02 07:36:35.19955196 +0000 UTC /home/build/src/java.base/unix/native/libnet/NetworkInterface.c:1304:5: warning: '__builtin_strncpy' output may be truncated copying 15 bytes from a string of length 15 [-Wstringop-truncation]
2025-06-02 07:36:35.199577702 +0000 UTC  1304 |     strncpy(if2.ifr_name, ifname, sizeof(if2.ifr_name) - 1);
2025-06-02 07:36:35.199595542 +0000 UTC       |     ^
2025-06-02 07:36:35.199626552 +0000 UTC /home/build/src/java.base/unix/native/libnet/NetworkInterface.c:1241:5: warning: '__builtin_strncpy' output may be truncated copying 15 bytes from a string of length 15 [-Wstringop-truncation]
2025-06-02 07:36:35.199651659 +0000 UTC  1241 |     strncpy(if2.ifr_name, name, sizeof(if2.ifr_name) - 1);
2025-06-02 07:36:35.199668784 +0000 UTC       |     ^
2025-06-02 07:36:35.199684864 +0000 UTC /home/build/src/java.base/unix/native/libnet/NetworkInterface.c:1241:5: warning: '__builtin_strncpy' output may be truncated copying 15 bytes from a string of length 15 [-Wstringop-truncation]
2025-06-02 07:36:35.684586936 +0000 UTC Compiling 30 files for java.security.sasl
2025-06-02 07:36:35.686516626 +0000 UTC Compiling 68 files for jdk.dynalink
2025-06-02 07:36:36.601398983 +0000 UTC Compiling 129 files for java.rmi
2025-06-02 07:36:36.771465944 +0000 UTC /home/build/src/java.base/share/native/libjimage/imageFile.cpp: In member function 'void ImageFileReader::location_path(ImageLocation&, char*, size_t) const':
2025-06-02 07:36:36.771618476 +0000 UTC /home/build/src/java.base/share/native/libjimage/imageFile.cpp:514:16: warning: 'char* __builtin_strncpy(char*, const char*, long unsigned int)' specified bound depends on the length of the source argument [-Wstringop-truncation]
2025-06-02 07:36:36.771643228 +0000 UTC   514 |         strncpy(next, parent, length); next += length;
2025-06-02 07:36:36.771661408 +0000 UTC       |                ^
2025-06-02 07:36:36.773635968 +0000 UTC /home/build/src/java.base/share/native/libjimage/imageFile.cpp:510:24: note: length computed here
2025-06-02 07:36:36.773758616 +0000 UTC   510 |         length = strlen(parent);
2025-06-02 07:36:36.77378335 +0000 UTC       |                  ~~~~~~^~~~~~~~
2025-06-02 07:36:36.773803472 +0000 UTC /home/build/src/java.base/share/native/libjimage/imageFile.cpp:524:12: warning: 'char* __builtin_strncpy(char*, const char*, long unsigned int)' specified bound depends on the length of the source argument [-Wstringop-truncation]
2025-06-02 07:36:36.773838811 +0000 UTC   524 |     strncpy(next, base, length); next += length;
2025-06-02 07:36:36.773863906 +0000 UTC       |            ^
2025-06-02 07:36:36.773972703 +0000 UTC /home/build/src/java.base/share/native/libjimage/imageFile.cpp:520:20: note: length computed here
2025-06-02 07:36:36.774004204 +0000 UTC   520 |     length = strlen(base);
2025-06-02 07:36:36.774021273 +0000 UTC       |              ~~~~~~^~~~~~
2025-06-02 07:36:38.74439604 +0000 UTC Compiling 251 files for jdk.jdi
2025-06-02 07:36:38.86453507 +0000 UTC Compiling 38 files for jdk.jcmd
2025-06-02 07:36:41.285699115 +0000 UTC make[2]: *** [make/Main.gmk:215: java.base-libs] Error 2
2025-06-02 07:36:41.285865113 +0000 UTC make[2]: *** Waiting for unfinished jobs....
2025-06-02 07:37:02.435767001 +0000 UTC 
2025-06-02 07:37:02.435930968 +0000 UTC ERROR: Build failed for target 'jdk-image' in configuration 'linux-x86_64-server-release' (exit code 2) 
2025-06-02 07:37:02.445475333 +0000 UTC Stopping sjavac server
2025-06-02 07:37:02.520374521 +0000 UTC 
2025-06-02 07:37:02.520513117 +0000 UTC === Output from failing command(s) repeated here ===
2025-06-02 07:37:02.522758049 +0000 UTC * For target support_native_java.base_libnet_DefaultProxySelector.o:
2025-06-02 07:37:02.528674766 +0000 UTC /home/build/src/java.base/unix/native/libnet/DefaultProxySelector.c: In function 'getProxyByGProxyResolver':
2025-06-02 07:37:02.528781128 +0000 UTC /home/build/src/java.base/unix/native/libnet/DefaultProxySelector.c:375:16: error: too many arguments to function 'g_proxy_resolver_lookup'; expected 0, have 4
2025-06-02 07:37:02.528806635 +0000 UTC   375 |     proxies = (*g_proxy_resolver_lookup)(resolver, uri, NULL, &error);
2025-06-02 07:37:02.528823259 +0000 UTC       |               ~^~~~~~~~~~~~~~~~~~~~~~~~~ ~~~~~~~~
2025-06-02 07:37:02.528862423 +0000 UTC /home/build/src/java.base/unix/native/libnet/DefaultProxySelector.c:394:34: error: too many arguments to function 'g_network_address_parse_uri'; expected 0, have 3
2025-06-02 07:37:02.528890355 +0000 UTC   394 |                                 (*g_network_address_parse_uri)(proxies[i], 0,
2025-06-02 07:37:02.528907985 +0000 UTC       |                                 ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ~~~~~~~~~~
2025-06-02 07:37:02.528923656 +0000 UTC /home/build/src/java.base/unix/native/libnet/DefaultProxySelector.c:399:38: error: too many arguments to function 'g_network_address_get_hostname'; expected 0, have 1
2025-06-02 07:37:02.528967891 +0000 UTC   399 |                             phost = (*g_network_address_get_hostname)(conn);
2025-06-02 07:37:02.528984317 +0000 UTC       |                                     ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ~~~~
2025-06-02 07:37:02.52899992 +0000 UTC /home/build/src/java.base/unix/native/libnet/DefaultProxySelector.c:400:38: error: too many arguments to function 'g_network_address_get_port'; expected 0, have 1
2025-06-02 07:37:02.529038443 +0000 UTC   400 |                             pport = (*g_network_address_get_port)(conn);
2025-06-02 07:37:02.541270783 +0000 UTC    ... (rest of output omitted)
2025-06-02 07:37:02.544038204 +0000 UTC 
2025-06-02 07:37:02.544143457 +0000 UTC * All command lines available in /home/build/build/linux-x86_64-server-release/make-support/failure-logs.
2025-06-02 07:37:02.548838746 +0000 UTC === End of repeated output ===
2025-06-02 07:37:02.569712172 +0000 UTC 
2025-06-02 07:37:02.569769276 +0000 UTC No indication of failed target found.
2025-06-02 07:37:02.571171617 +0000 UTC Hint: Try searching the build log for '] Error'.
2025-06-02 07:37:02.577404931 +0000 UTC make[1]: *** [/home/build/make/Init.gmk:310: main] Error 2
2025-06-02 07:37:02.577411577 +0000 UTC Hint: See doc/building.html#troubleshooting for assistance.
2025-06-02 07:37:02.577446462 +0000 UTC make: *** [/home/build/make/Init.gmk:186: jdk-image] Error 2
2025-06-02 07:37:02.577459324 +0000 UTC 
2025-06-02 07:37:10.024780096 +0000 UTC failed to build package: unable to run package openjdk-12 pipeline: unable to run pipeline: exit status 2
2025-06-02 07:37:10.059949316 +0000 UTC failed to build: command melange failed: exit status 1
2025-06-02 07:37:10.06351216 +0000 UTC build failed: command melange failed: exit status 1
```

--- a/openjdk-12.yaml
+++ b/openjdk-12.yaml
@@ -129,6 +129,7 @@ subpackages:
         - libfontconfig1
         - libjpeg-turbo
         - libxext
+        - libxi
         - libxrender
         - libxtst
         - openjdk-12-jre-base

--- a/openjdk-12.yaml
+++ b/openjdk-12.yaml
@@ -1,7 +1,7 @@
 package:
   name: openjdk-12
   version: 12.0.2.10
-  epoch: 7
+  epoch: 8
   description:
   copyright:
     - license: GPL-2.0-with-classpath-exception
@@ -24,6 +24,7 @@ environment:
       - fontconfig-dev
       - freetype-dev
       - gawk
+      - gcc-14-default
       - giflib-dev
       - lcms2-dev
       - libffi-dev

